SYCL/ESIMD/api/functional/type_coverage.hpp

```diff
@@ -34,22 +34,29 @@ template <typename... T> struct named_type_pack {
       : names{std::forward<nameListT>(nameList)...} {}
 };
 
-enum class tested_types { all, fp, uint, sint };
+// Provides alias to types that can be used in tests:
+//  core - all C++ data types, except specific data types
+//  fp - all floating point C++ data types
+//  fp_extra - specific, non C++ data types
+//  uint - all unsigned C++ integral data types
+//  sint - all signed C++ integral data types
+enum class tested_types { core, fp, fp_extra, uint, sint };
 
 // Factory method to retrieve pre-defined named_type_pack, to have the same
 // default type coverage over the tests
 template <tested_types required> auto get_tested_types() {
-  if constexpr (required == tested_types::all) {
+  if constexpr (required == tested_types::core) {
     return named_type_pack<char, unsigned char, signed char, short,
                            unsigned short, int, unsigned int, long,
-                           unsigned long, float, sycl::half, double, long long,
+                           unsigned long, float, double, long long,
                            unsigned long long>(
         {"char", "unsigned char", "signed char", "short", "unsigned short",
-         "int", "unsigned int", "long", "unsigned long", "float", "sycl::half",
-         "double", "long long", "unsigned long long"});
+         "int", "unsigned int", "long", "unsigned long", "float", "double",
```

Comment:
I think 'double' is also one of 'extra_fp' types as some devices may require sycl::aspect checks before running 'double' features.

Reply:
I've done it in this commit: 17555715d874898214e5d6c3b0ffeb5e511408f1